### DISCUSSION TITLE
How Do I Merge a Branch?

### DISCUSSION THREAD

**Discussion by dev-rygy** [community/community#28250] [New to GitHub] [answered]
Hey I'm currently working on a collaborative project with my friend and we are having a bit of trouble understanding how to merge a GitHub branch. What is a pull request? How do I resolve conflicts? Really I'm just looking for an overview of the process on how a correct merge is done.

**Comment by atuschek0** [comment, marked as answer]
Okay, so first before you merge your branch, commit all your recent changes. Then, on the web version of Github, make a pull request from the branch you're wanting to merge. 

In the pull request select what you want to merge your branch to, main or an other branch. It's good practice to write a good description stating how the branch is going to change the project. After that click on "Create Pull Request". If you're working alone and there are no conflicts then you can go ahead and click on merge, but it is still good to review all the changes you've made in the "files changed" tab before you merge. 
If you are working with someone else and you are not an admin of the repository you may need their approval first. If you have a conflict you will need to resolve it with your team you can either keep their changes (the source) or your changes (the incoming). You can also cherry pick the conflicts on the web app. Once you have resolved the conflicts Github will turn green and you are ready to
merge :)

After that you have successfully merged to main/anotherBranch!!

**Reply by queenofcorgis** [reply]
Thank you @atuschek0 for such a detailed reply ❤️ 

And thank you @MaDness1184 for marking the answer ✅
